lib/travis/api/app/endpoint/repos.rb

```diff
@@ -16,8 +16,13 @@ class Repos < Endpoint
       # json(:repositories)
       get '/' do
         prefer_follower do
-          params['ids'] = params['ids'].split(',') if params['ids'].respond_to?(:split)
-          respond_with service(:find_repos, params)
+          find_repos_service = service(:find_repos, params).run
+          if find_repos_service
+            params['ids'] = params['ids'].split(',') if params['ids'].respond_to?(:split)
```

Comment:
Hey Igor, thanks for reviewing this! 
I guess a 403 makes way more sense.
I tried to follow what is done on the `repo/:id/settings` endpoint https://github.com/travis-ci/travis-api/blob/master/lib/travis/api/app/endpoint/repos.rb#L53  to keep the code uniform in a certain way. I can change the logic to make it clearer this is an endpoint only for authenticated users.